src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -484,12 +484,16 @@ private static List<ChatMessageContentItem> GetContentParts(IList<AIContent> con
                     parts.Add(new ChatMessageTextContentItem(textContent.Text));
                     break;
 
-                case ImageContent imageContent when imageContent.Data is { IsEmpty: false } data:
-                    parts.Add(new ChatMessageImageContentItem(BinaryData.FromBytes(data), imageContent.MediaType));
-                    break;
+                case DataContent dataContent when dataContent.HasImageMediaType():
+                    if (dataContent.Data is { IsEmpty: false } data)
```

Comment:
This is fine, but fwiw I find the IsEmpty: false double-negative a little confusing. This could also be:
```C#
if (dataContent.ContainsData)
{
    ... dataContent.Data ...
}
```

Reply:
I usually go for `is not []` but this requires polyfilling `System.Index` in netstandard2.0